csrc/logical_domain_map.cpp

```diff
@@ -361,35 +361,44 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
   }
 
   if (auto* op = dynamic_cast<SdpaFwdOp*>(consumer_tv_->definition())) {
-    // Note: Explicit handling of DIDx(D) until
-    // https://github.com/NVIDIA/Fuser/issues/2563 is resolved. Producers:
-    //   query = [DIDx(D)?, N, H, L, E]
-    //   key = [DIDx(D)?, N, H, S, E]
-    //   value = [DIDx(D)?, N, H, S, Ev]
+    // Producers:
+    //   query = [N*, L, E]
+    //   key = [N*, S, E]
+    //   value = [N*, S, Ev]
+    //
     // Consumers:
-    //   output = [DIDx(D)?, N, H, L, Ev]
-    //   logsumexp = [DIDx(D)?, N, H, L]
-    // Note: S, E are not mapped together in the producers and do not have any
+    //   output = [N*, L, Ev]
+    //   logsumexp = [N*, L]
+    //   philox_seed = []
+    //   philox_offset = []
+    //
+    // N* are the dimensions that are treated batch, e.g., the actual batch
+    // dimension, the head dimension and the extra sequence dimension in
+    // [Triangle
+    // Attention](https://elanapearl.github.io/blog/2024/the-illustrated-alphafold/#triangle-attention).
+    //
+    // S, E are not mapped together in the producers and do not have any
     // mapping to the consumer.
-    if (consumer_tv_->sameAs(op->philox_seed())) {
+    if (consumer_tv_ == op->philox_seed() ||
+        consumer_tv_ == op->philox_offset()) {
       return dom_map;
     }
-    size_t num_device_dim = producer_logical.at(0)->isDeviceDim() ? 1 : 0;
-    // Map N, H from any input (query/key/value)
-    for (auto idx : arange(consumer_root.size())) {
-      if (idx < (2 + num_device_dim)) {
-        updatePairwiseLogicalDomainMap(
-            producer_logical.at(idx), consumer_root.at(idx));
+
+    const auto num_batch_dims =
+        std::ssize(op->attn_out()->getLogicalDomain()) - 2;
```

Comment:
```suggestion
        static_cast<std::size_t>(std::ssize(op->attn_out()->getLogicalDomain()) - 2);
```
**[Suggested by AI]** Summary of the fix

Problem  
• std::ssize(...) returns a signed type (typically std::ptrdiff_t).  
• The result was later being used where an unsigned type (std::size_t) was expected, which produced a build/​conversion error or warning.

Change made  
• Wrapped the result of the expression in a static_cast to std::size_t:

    static_cast<std::size_t>(std::ssize(op->attn_out()->getLogicalDomain()) - 2);

Effect  
• Forces an explicit, well-defined conversion from the signed value to an unsigned size_t.  
• Eliminates the compiler error/warning about implicit conversion or type mismatch, allowing the code to build successfully.